MainPerson/src/com/beyond/person/core/service/implementation/DancerImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.beyond.person.core.service.implementation;
+import com.beyond.person.core.model.Dancer;
+import com.beyond.person.core.service.interfaces.DancerAction;
+
+public class DancerImpl implements DancerAction {
+    Dancer dancer;
+
+    /**
+     * constructor for Dancer Impl
+     * @param name
+     * @param designation
+     * @param groupName
+     */
+    public DancerImpl(String name, String designation, String groupName) {
+       dancer = new Dancer(name, designation, groupName);
+    }
+
+
+    /**
+     * method dancing only for dancer from interface DancerAction
+     */
+    @Override
+    public void dancing() {
+        System.out.println(dancer.getName() + " dances at "+  dancer.getGroupName());
+    }
+
+    /**
+     * eat overriden
+     */
+    @Override
+    public void eat() {
+        System.out.println(dancer.getName() + " is vegetarian");
+
+    }
+
+    /**
+     * learn overriden
+     */
+    @Override
+    public void learn() {
+        System.out.println( dancer.getName() + " has a dancing Education");
+
+    }
+
+    /**
+     * all methods of dancer in one
+     */
+    @Override
+    public void all() {
```

Comment:
Try to make all available for base class but at the same time add in each exact impl specific behavior

Reply:
it is available for the base class, it is a method in basepersonAction, and the other interfaces take this method from the basePersonAction, and each implement a specific behavior